modules/db/src/main/java/org/mule/module/db/internal/domain/connection/AbstractConnectionFactory.java

```diff
@@ -17,14 +18,25 @@
 public abstract class AbstractConnectionFactory implements ConnectionFactory
 {
 
+    /**
+     * Ensures DriverManager classloading takes place before any connection creation.
+     * It prevents a JDK deadlock that only occurs when two JDBC Connections of different DB vendors
+     * are created concurrently and the {@link DriverManager} hasn't been loaded yet.
+     * For more information, see MULE-14605.
+     */
+    static
+    {
+        DriverManager.getLoginTimeout();
```

Comment:
If I'm understanding this correctly, this static block will be executed the first time the AbstractConnectionFactory class is referenced. If there are two applications, each having their own drivers, then the problem still exists as the initialization will occur during on the first app, but not in the second.